### PR TITLE
UX-680/Multiversion support on both create and upgrade

### PR DESCRIPTION
**Pull request by jruddell**
Only one upgrade option
image

multiple upgrade options
image

cluster create
image

### CODE REVIEW

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ClusterUpgradeDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+import Text from 'core/elements/text'
+import { FormControlLabel, Radio, RadioGroup } from '@material-ui/core'
+
+enum UpgradeTypes {
+  Minor = 'minor',
+  Patch = 'patch',
+}
+
+function getUpgradeTarget(cluster) {
+  if ((cluster.canMinorUpgrade && cluster.canPatchUpgrade) || cluster.canMinorUpgrade) {
```
Comment (kevinylee1991) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/ClusterUpgradeDialog.tsx:14]: The conditions seems redundant, seems the first one can be removed?

Comment (knnguy) [inline src/app/plugins/kubernetes/components/infrastructure/clusters/ClusterUpgradeDialog.tsx:14]: Isn't the `cluster.canPatchUpgrade ` here redundant?

**Review by kevinylee1991 — APPROVED** [review]

File: src/app/plugins/kubernetes/components/infrastructure/clusters/ClusterUpgradeDialog.tsx (hunk 1 above)
Comment (jruddell) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/ClusterUpgradeDialog.tsx:14]: so in both cases of 2 possible upgrades or if there is only one possible upgrade of type `minor` i want to default to a minor upgrade.

Comment (knnguy) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/ClusterUpgradeDialog.tsx:14]: So you would only need `if (cluster.canMinorUpgrade)`. That's what I was trying to say

Comment (jruddell) [reply src/app/plugins/kubernetes/components/infrastructure/clusters/ClusterUpgradeDialog.tsx:14]: sorry misread what you were saying. I was thinking it was a bit more explicit but, it probably reads better to do it that way

File: src/api-client/ApiClient.ts [hunk 2]
```diff
@@ -175,88 +175,93 @@ class ApiClient {
...
     url,
+    version,
     endpoint = undefined,
     config = undefined,
     options: { clsName, mthdName },
   }: IRawRequestGetParams) => {
-    if (endpoint === undefined) {
-      endpoint = await this.apiServices[clsName].getApiEndpoint()
-    }
+    endpoint = await this.getEndpoint({ endpoint, version, clsName })
```
Comment (xantorres) [inline src/api-client/ApiClient.ts:195]: A decorator would have done the trick in a more elegant way

**Review by xantorres — APPROVED** [review]

**Pull request merged**